### PR TITLE
Update gradle 7 and remove aspectj (#21684)

### PR DESCRIPTION
**Pull request by spbolton**
Need a good amount of testing on this before and after merging.  Dependencies have been updated, upgrade of gradle and some of the conventions.  Using bytebuddy to handle DB and transaction annotations.  ByteBuddy agent string removed from batch files as java 11 it can be loaded dynamically.   Please ask if any questions about the changes and any impact.

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

File: dotCMS/build.gradle [hunk 1]
```diff
@@ -253,62 +346,319 @@ ext.prepareFelixExternalFolder = { felixExternalFolder ->
...
+
+    // Copying libs/buildlibs under tomcat8/lib.
+    def copyBuildLibs(buildLibLocation) {
+
+        project.copy {
+            from project.configurations.compileClasspath
+            into buildLibLocation + '/lib'
+            include '**/ojdbc*.jar'
+            include '**/postgresql*.jar'
+            include '**/mysql-connector-java*.jar'
```
Comment (jdotcms) [inline dotCMS/build.gradle:448]: do we still needing Oracle and mysql?

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import net.bytebuddy.dynamic.ClassFileLocator;
+import net.bytebuddy.matcher.ElementMatchers;
+import net.bytebuddy.utility.JavaModule;
+import org.apache.logging.log4j.Level;
+
+import java.lang.instrument.Instrumentation;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+public class ByteBuddyFactory {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:26]: Doc me

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        init();
+    }
+
+    private void init() {
+        try {
+            premain(null, ByteBuddyAgent.install());
+            Logger.info(ByteBuddyFactory.class, "Loaded ByteBuddy Advice");
+
+        } catch (Exception e) {
+            Logger.error(ByteBuddyFactory.class, "Cannot install ByteBuddy Advice");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:43]: I would log the exception such as
Logger.debug(

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        }
+    }
+
+    public static void main(String[] args) {
+        ByteBuddyFactory inst = ByteBuddyFactory.getInstance();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:48]: do we need this main

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        } catch (DotSecurityException e) {
+            e.printStackTrace();
+        }
+    }
+    public static ByteBuddyFactory getInstance() {
+        return INSTANCE;
+    }
+
+
+    public void premain(String arg, Instrumentation inst)  throws Exception  {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:72]: set params to final

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        Logger.info(ByteBuddyFactory.class,"ByteBuddy Initialized");
+    } catch (Exception e)
+    {
+        Logger.error(ByteBuddyFactory.class, "Error Initializing ByteBuddy",e);
+    }
+
+    }
+
+    private static <U extends MethodDescription> boolean matchTest(U a) {
+        Logger.error(ByteBuddyFactory.class, "Tester=" + a.getDeclaringType().getTypeName() + ":" + a.getName());
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:146]: is this method needed?

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        return true;
+    }
+
+
+    private boolean checkLogLevelRequired(MethodDescription target) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:151]: set to final

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionAdvice.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.github.rjeschke.txtmark.Run;
+import net.bytebuddy.asm.Advice;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import java.lang.reflect.Method;
+import java.sql.Connection;
+
+public class WrapInTransactionAdvice {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionAdvice.java:16]: I am not sure if this advicer is handling the externalize=true

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionInterceptor.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import static com.dotcms.util.AnnotationUtils.getMethodAnnotation;
+
+public class WrapInTransactionInterceptor {
+
+    @RuntimeType
+    // Better to rewrite as an Advice
+    public static Object intercept(@SuperCall Callable<?> delegate, @Origin Method method) throws Exception {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionInterceptor.java:19]: Oh I see the externalize is here, nvm previous msg

Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionInterceptor.java:19]: No In fact this Intercepter is no longer used and has been replaced as the Advice so my comment to the previous message does apply and this should be removed.   The advice should be more efficient.

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/MethodHookAdvice.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
+package com.dotcms.business.bytebuddy;
+
+import com.dotmarketing.db.DbConnectionFactory;
+import net.bytebuddy.asm.Advice;
+
+public class MethodHookAdvice {
+
+    @Advice.OnMethodExit(inline = false, onThrowable = Throwable.class)
+    public static void exit(
+            @Advice.Thrown Throwable t
```
[the remark this answers is not in the document]
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/MethodHookAdvice.java:10]: Yeah this was a test class I will clean up.

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionAdvice.java (hunk 8 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionAdvice.java:16]: Create a new annotation would be ok

File: dotCMS/build.gradle (hunk 1 above)
Comment (spbolton) [reply dotCMS/build.gradle:448]: Question for @wezell Do we want to clean these up now or still intending on working with them for the time being?

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java (hunk 7 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java]: done

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionAdvice.java (hunk 8 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionAdvice.java:16]: I this something we need at the moment for the system to work or can we add a task for this?

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionInterceptor.java (hunk 9 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/WrapInTransactionInterceptor.java]: I have now cleaned up this class

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/MethodHookAdvice.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/MethodHookAdvice.java]: Now cleaned up

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java (hunk 5 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java]: Done

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java (hunk 4 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java]: Removed

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java (hunk 3 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:48]: Why debug?,  if this happens it is a fatal error and the system will fail so should be an error.

File: dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java (hunk 2 above)
Comment (spbolton) [reply dotCMS/src/main/java/com/dotcms/business/bytebuddy/ByteBuddyFactory.java:34]: done

**Review by nollymar — CHANGES_REQUESTED** [review]
I see many tests broken. It might need the latest changes from master

**Review by jdotcms — APPROVED** [review]

**Pull request merged**